### PR TITLE
74322 api  get preservation record endpoint

### PR DESCRIPTION
**Pull request by demissie**

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                return new NotFoundResult<PreservationRecordDto>(Strings.EntityNotFound(nameof(Tag), request.TagId));
+            }
+
+            var requirement = tag.Requirements.SingleOrDefault(r => r.Id == request.RequirementId);
+            if (requirement == null)
+            {
+                return new NotFoundResult<PreservationRecordDto>(Strings.EntityNotFound(nameof(TagRequirement), request.RequirementId));
+            }
+
+            var records = requirement.PreservationPeriods.Select(pp => pp.PreservationRecord);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs:47]: exclude periods without records

File: src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            var preservationRecord = records.SingleOrDefault(pr => pr.Id == request.PreservationRecordId);
+            if (preservationRecord == null)
+            {
+                return new NotFoundResult<PreservationRecordDto>(Strings.EntityNotFound(nameof(PreservationPeriod), request.PreservationRecordId));
+            }
+
+            // get needed information about preservation record
+            var preservationRecordDto =
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs:55]: remove from line 55 to 65

File: src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        PresRecordRowVersion = pr.RowVersion.ToString()
+                    });
+
+            var requirementDto =
+                preservationRecordDto.Single(rd => rd.PreservationRecordId == request.PreservationRecordId);
+
+            var preservRecordDto = new PreservationRecordDto(
+                preservationRecord.Id,
+                preservationRecord.BulkPreserved,
+                requirementDto.PresRecordRowVersion);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs:71]: no need for RowVersion in this dto

File: src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+            return new SuccessResult<PreservationRecordDto>(preservRecordDto);
+        }
+
+        private class Dto
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/GetPreservationRecordQueryHandler.cs:76]: remove

File: src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/PreservationRecordDto.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            string rowVersion)
+        {
+            Id = id;
+            BulkPreserved = bulkPreserved;
+            RowVersion = rowVersion;
+        }
+
+        public int Id { get; }
+        public bool BulkPreserved { get; }
+        public string RowVersion { get; }
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetPreservationRecord/PreservationRecordDto.cs:17]: remove

File: src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs [hunk 6]
```diff
@@ -90,7 +91,7 @@ public class TagsController : ControllerBase
 
         [Authorize(Roles = Permissions.PRESERVATION_READ)]
         [HttpGet("{id}/Requirements")]
-        public async Task<ActionResult<List<RequirementDto>>> GetTagRequirements(
+        public async Task<ActionResult<List<TagRequirementDto>>> GetTagRequirements(
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs:94]: This is a breaking change! Revert and take in using RequirementDto = Equinor.Procosys.Preservation.Query.GetTagRequirements.RequirementDto; at line 46 again

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        protected override void SetupNewDatabase(DbContextOptions<PreservationContext> dbContextOptions)
+        {
+            using (var context = new PreservationContext(dbContextOptions, _plantProvider, _eventDispatcher,
+                _currentUserProvider))
+            {
+                var journey = AddJourneyWithStep(context, "J1", "S", AddMode(context, "M1", false), AddResponsible(context, "R1"));
+                var reqDef = new RequirementDefinition(TestPlant, "TestTitle", 2, RequirementUsage.ForAll, 1);
+
+                var requirement = new TagRequirement(TestPlant, 2, reqDef);
+                context.Add(requirement);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs:34]: can't do this. Will be saved when saving tag

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+                context.SaveChangesAsync().Wait();
+                _requirementId = requirement.Id;
+
+                var preservationRecord = new PreservationRecord(TestPlant, new Mock<Person>().Object, true);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs:38]: can't do this!
A TagRequirement get a PreservationPeriod when starting preservation: Start preseration on Tag.
A PreservationPeriod get a PreservationRecord when preserving: Preserve the Tag

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+namespace Equinor.Procosys.Preservation.Query.Tests.GetPreservationRecord
+{
+    [TestClass]
+    public class GetGetPreservationRecordQueryHandlerTests : ReadOnlyTestsBase
+    {
+        private int _tagId;
+        private int _requirementId;
+        private  int _preservationRecordId;
+
+        //private readonly DateTime _dueTimeUtc = DateTime.UtcNow;
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetPreservationRecord/GetPreservationRecordQueryHandlerTests.cs:23]: remove

File: src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs [hunk 10]
```diff
@@ -43,6 +44,7 @@
 using ServiceResult;
 using ServiceResult.ApiExtensions;
 using RequirementDto = Equinor.Procosys.Preservation.Query.GetTagRequirements.RequirementDto;
+using PreservationRecordDto = Equinor.Procosys.Preservation.Query.GetPreservationRecord.PreservationRecordDto;
```
Comment (llykkelarsen) [inline src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs:47]: Don't think this is necessary, but probably not worth the time to remove as this is my only comment.

**Pull request merged**